# HG changeset patch
# Parent eafecdfd37d55c9f18d051ca2f15005944f4946f
# User Frédéric Wang <fred.wang@free.fr>
MathML Operator dictionary: fix conflicts and add entries for bars and quadruple arrows (bug 414294). r=karlt

diff --git a/layout/mathml/mathfont.properties b/layout/mathml/mathfont.properties
--- a/layout/mathml/mathfont.properties
+++ b/layout/mathml/mathfont.properties
@@ -132,17 +132,17 @@ operator.\u002C.infix = lspace:0 rspace:
 operator.\u002D.infix = lspace:4 rspace:4 # -
 operator.\u002D.prefix = lspace:0 rspace:1 # -
 operator.\u002D\u002D.postfix = lspace:0 rspace:0 # --
 operator.\u002D\u003D.infix = lspace:4 rspace:4 # -=
 operator.\u002D\u003E.infix = lspace:5 rspace:5 # ->
 operator.\u002E.infix = lspace:3 rspace:3 # .
 operator.\u002E\u002E.postfix = lspace:0 rspace:0 # ..
 operator.\u002E\u002E\u002E.postfix = lspace:0 rspace:0 # ...
-operator.\u002F.infix = lspace:3 rspace:3 stretchy direction:vertical # /
+operator.\u002F.infix = lspace:1 rspace:1 direction:vertical # solidus
 operator.\u002F\u002F.infix = lspace:1 rspace:1 # //
 operator.\u002F\u003D.infix = lspace:4 rspace:4 # /=
 operator.\u003A.infix = lspace:1 rspace:2 # :
 operator.\u003A\u003D.infix = lspace:4 rspace:4 # :=
 operator.\u003B.infix = lspace:0 rspace:3 separator # ;
 operator.\u003C.infix = lspace:5 rspace:5 # &lt;
 operator.\u003C\u003D.infix = lspace:5 rspace:5 # &lt;=
 operator.\u003C\u003E.infix = lspace:1 rspace:1 # &lt;>
@@ -153,23 +153,23 @@ operator.\u003E\u003D.infix = lspace:5 r
 operator.\u003F.infix = lspace:1 rspace:1 # ?
 operator.\u0040.infix = lspace:1 rspace:1 # @
 operator.\u005B.prefix = lspace:0 rspace:0 stretchy fence symmetric direction:vertical mirrorable # [
 operator.\u005C.infix = lspace:0 rspace:0 # reverse solidus
 operator.\u005D.postfix = lspace:0 rspace:0 stretchy fence symmetric direction:vertical mirrorable # ]
 operator.\u005E.postfix = lspace:0 rspace:0 stretchy accent direction:horizontal # &Hat; circumflex accent
 operator.\u005E.infix = lspace:1 rspace:1 direction:horizontal # ^
 operator.\u005F.postfix = lspace:0 rspace:0 stretchy accent direction:horizontal # _ low line
-operator.\u005F.infix = lspace:2 rspace:2 stretchy direction:horizontal # _
+operator.\u005F.infix = lspace:1 rspace:1 direction:horizontal # _ low line
 operator.\u0060.postfix = lspace:0 rspace:0 accent # &DiacriticalGrave;
 operator.\u007B.prefix = lspace:0 rspace:0 stretchy fence symmetric direction:vertical mirrorable # {
 operator.\u007C.infix = lspace:2 rspace:2 stretchy fence symmetric direction:vertical # &VerticalLine; |
 operator.\u007C.prefix = lspace:0 rspace:0 stretchy fence symmetric direction:vertical # |
 operator.\u007C.postfix = lspace:0 rspace:0 stretchy fence symmetric direction:vertical # |
-operator.\u007C\u007C.infix = lspace:4 rspace:4 direction:vertical # ||
+operator.\u007C\u007C.infix = lspace:2 rspace:2 stretchy fence symmetric direction:vertical # ||
 operator.\u007C\u007C.prefix = lspace:0 rspace:0 stretchy fence symmetric direction:vertical # multiple character operator: ||
 operator.\u007C\u007C.postfix = lspace:0 rspace:0 stretchy fence symmetric direction:vertical # multiple character operator: ||
 operator.\u007C\u007C\u007C.infix = lspace:2 rspace:2 stretchy fence symmetric direction:vertical # multiple character operator: |||
 operator.\u007C\u007C\u007C.prefix = lspace:0 rspace:0 stretchy fence symmetric direction:vertical # multiple character operator: |||
 operator.\u007C\u007C\u007C.postfix = lspace:0 rspace:0 stretchy fence symmetric direction:vertical # multiple character operator: |||
 operator.\u007D.postfix = lspace:0 rspace:0 stretchy fence symmetric direction:vertical mirrorable # }
 operator.\u007E.postfix = lspace:0 rspace:0 stretchy accent direction:horizontal # ~ tilde
 operator.\u00A8.postfix = lspace:0 rspace:0 accent # &DoubleDot;
@@ -203,16 +203,17 @@ operator.\u2016.postfix = lspace:0 rspac
 operator.\u2018.prefix = lspace:0 rspace:0 fence # &OpenCurlyQuote;
 operator.\u2019.postfix = lspace:0 rspace:0 fence # &CloseCurlyQuote;
 operator.\u201C.prefix = lspace:0 rspace:0 fence # &OpenCurlyDoubleQuote;
 operator.\u201D.postfix = lspace:0 rspace:0 fence # &CloseCurlyDoubleQuote;
 operator.\u2022.infix = lspace:4 rspace:4 # bullet
 operator.\u2026.infix = lspace:0 rspace:0 # horizontal ellipsis
 operator.\u2032.postfix = lspace:0 rspace:2 # prime
 operator.\u203E.postfix = lspace:0 rspace:0 stretchy accent direction:horizontal # overline
+operator.\u2044.infix = lspace:4 rspace:4 stretchy # fraction slash
 operator.\u2061.infix = lspace:0 rspace:0 # &ApplyFunction;
 operator.\u2062.infix = lspace:0 rspace:0 # &InvisibleTimes;
 operator.\u2063.infix = lspace:0 rspace:0 separator # &InvisibleComma;
 operator.\u2064.infix = lspace:0 rspace:0 # invisible plus
 operator.\u20DB.postfix = lspace:0 rspace:0 accent # &TripleDot;
 operator.\u20DC.postfix = lspace:0 rspace:0 accent # combining four dots above
 operator.\u2145.prefix = lspace:2 rspace:1 # &CapitalDifferentialD;
 operator.\u2146.prefix = lspace:2 rspace:0 # &DifferentialD;
@@ -341,31 +342,32 @@ operator.\u220E.infix = lspace:3 rspace:
 operator.\u220F.prefix = lspace:1 rspace:2 largeop movablelimits symmetric direction:vertical # &Product;
 operator.\u2210.prefix = lspace:1 rspace:2 largeop movablelimits symmetric direction:vertical # &Coproduct;
 operator.\u2211.prefix = lspace:1 rspace:2 largeop movablelimits symmetric direction:vertical mirrorable # &Sum;
 operator.\u2212.infix = lspace:4 rspace:4 # official Unicode minus sign
 operator.\u2212.prefix = lspace:0 rspace:1 # official Unicode minus sign
 operator.\u2213.infix = lspace:4 rspace:4 # &MinusPlus;
 operator.\u2213.prefix = lspace:0 rspace:1 # &MinusPlus;
 operator.\u2214.infix = lspace:4 rspace:4 # dot plus
-operator.\u2216.infix = lspace:3 rspace:3 stretchy direction:vertical # &Backslash;
+operator.\u2215.infix = lspace:4 rspace:4 stretchy # division slash
+operator.\u2216.infix = lspace:4 rspace:4 direction:vertical # set minus
 operator.\u2217.infix = lspace:4 rspace:4 # asterisk operator
 operator.\u2218.infix = lspace:4 rspace:4 # &SmallCircle;
 operator.\u2219.infix = lspace:4 rspace:4 # bullet operator
 operator.\u221A.prefix = lspace:1 rspace:1 stretchy direction:vertical mirrorable # &Sqrt;
 operator.\u221B.prefix = lspace:1 rspace:1 # cube root
 operator.\u221C.prefix = lspace:1 rspace:1 # fourth root
 operator.\u221D.infix = lspace:5 rspace:5 # &Proportional;
 operator.\u221F.infix = lspace:5 rspace:5 # right angle
 operator.\u2220.prefix = lspace:0 rspace:0 # angle
 operator.\u2221.prefix = lspace:0 rspace:0 # measured angle
 operator.\u2222.prefix = lspace:0 rspace:0 # spherical angle
-operator.\u2223.infix = lspace:5 rspace:5 stretchy direction:vertical # &VerticalBar;
+operator.\u2223.infix = lspace:5 rspace:5 direction:vertical # divides
 operator.\u2224.infix = lspace:5 rspace:5 # &NotVerticalBar;
-operator.\u2225.infix = lspace:5 rspace:5 stretchy direction:vertical # &DoubleVerticalBar;
+operator.\u2225.infix = lspace:5 rspace:5 direction:vertical # parallel to
 operator.\u2226.infix = lspace:5 rspace:5 # &NotDoubleVerticalBar;
 operator.\u2227.infix = lspace:4 rspace:4 # &wedge;
 operator.\u2228.infix = lspace:4 rspace:4 # &vee;
 operator.\u2229.infix = lspace:4 rspace:4 # &cap;
 operator.\u222A.infix = lspace:4 rspace:4 # &cup;
 operator.\u222B.prefix = lspace:0 rspace:1 largeop symmetric direction:vertical integral mirrorable # &Integral;
 operator.\u222C.prefix = lspace:0 rspace:1 largeop symmetric direction:vertical integral mirrorable # double integral
 operator.\u222D.prefix = lspace:0 rspace:1 largeop symmetric direction:vertical integral mirrorable # triple integral
@@ -638,29 +640,31 @@ operator.\u25CD.infix = lspace:4 rspace:
 operator.\u25CE.infix = lspace:4 rspace:4 # bullseye
 operator.\u25CF.infix = lspace:4 rspace:4 # black circle
 operator.\u25D6.infix = lspace:4 rspace:4 # left half black circle
 operator.\u25D7.infix = lspace:4 rspace:4 # right half black circle
 operator.\u25E6.infix = lspace:4 rspace:4 # white bullet
 operator.\u266D.postfix = lspace:0 rspace:2 # music flat sign
 operator.\u266E.postfix = lspace:0 rspace:2 # music natural sign
 operator.\u266F.postfix = lspace:0 rspace:2 # music sharp sign
-operator.\u2758.infix = lspace:5 rspace:5 stretchy symmetric direction:vertical # &VerticalSeparator;
+operator.\u2758.infix = lspace:5 rspace:5 direction:vertical # light vertical bar 
 operator.\u2772.prefix = lspace:0 rspace:0 stretchy fence symmetric direction:vertical # light left tortoise shell bracket ornament
 operator.\u2773.postfix = lspace:0 rspace:0 stretchy fence symmetric direction:vertical # light right tortoise shell bracket ornament
 operator.\u27E6.prefix = lspace:0 rspace:0 stretchy fence symmetric direction:vertical # &LeftDoubleBracket;
 operator.\u27E7.postfix = lspace:0 rspace:0 stretchy fence symmetric direction:vertical # &RightDoubleBracket;
 operator.\u27E8.prefix = lspace:0 rspace:0 stretchy fence symmetric direction:vertical # &LeftAngleBracket;
 operator.\u27E9.postfix = lspace:0 rspace:0 stretchy fence symmetric direction:vertical # &RightAngleBracket;
 operator.\u27EA.prefix = lspace:0 rspace:0 stretchy fence symmetric direction:vertical # mathematical left double angle bracket
 operator.\u27EB.postfix = lspace:0 rspace:0 stretchy fence symmetric direction:vertical # mathematical right double angle bracket
 operator.\u27EC.prefix = lspace:0 rspace:0 stretchy fence symmetric direction:vertical # mathematical left white tortoise shell bracket
 operator.\u27ED.postfix = lspace:0 rspace:0 stretchy fence symmetric direction:vertical # mathematical right white tortoise shell bracket
 operator.\u27EE.prefix = lspace:0 rspace:0 stretchy fence symmetric direction:vertical # mathematical left flattened parenthesis
 operator.\u27EF.postfix = lspace:0 rspace:0 stretchy fence symmetric direction:vertical # mathematical right flattened parenthesis
+operator.\u27F0.infix = lspace:5 rspace:5 stretchy direction:vertical # upwards quadruple arrow
+operator.\u27F1.infix = lspace:5 rspace:5 stretchy direction:vertical # downwards quadruple arrow
 operator.\u27F5.infix = lspace:5 rspace:5 stretchy accent direction:horizontal # &LongLeftArrow;
 operator.\u27F6.infix = lspace:5 rspace:5 stretchy accent direction:horizontal # &LongRightArrow;
 operator.\u27F7.infix = lspace:5 rspace:5 stretchy accent direction:horizontal # &LongLeftRightArrow;
 operator.\u27F8.infix = lspace:5 rspace:5 stretchy accent direction:horizontal # &DoubleLongLeftArrow;
 operator.\u27F9.infix = lspace:5 rspace:5 stretchy accent direction:horizontal # &DoubleLongRightArrow;
 operator.\u27FA.infix = lspace:5 rspace:5 stretchy accent direction:horizontal # &DoubleLongLeftRightArrow;
 operator.\u27FB.infix = lspace:5 rspace:5 stretchy accent direction:horizontal # long leftwards arrow from bar
 operator.\u27FC.infix = lspace:5 rspace:5 stretchy accent direction:horizontal # long rightwards arrow from bar
@@ -1146,16 +1150,17 @@ operator.\u2AD5.infix = lspace:5 rspace:
 operator.\u2AD6.infix = lspace:5 rspace:5 # superset above superset
 operator.\u2AD7.infix = lspace:5 rspace:5 # superset beside subset
 operator.\u2AD8.infix = lspace:5 rspace:5 # superset beside and joined by dash with subset
 operator.\u2AD9.infix = lspace:5 rspace:5 # element of opening downwards
 operator.\u2ADA.infix = lspace:5 rspace:5 # pitchfork with tee top
 operator.\u2ADB.infix = lspace:5 rspace:5 # transversal intersection
 operator.\u2ADC.infix = lspace:5 rspace:5 # forking
 operator.\u2ADD.infix = lspace:5 rspace:5 # nonforking
+operator.\u2ADD\u0338.infix = lspace:5 rspace:5 # nonforking with slash
 operator.\u2ADE.infix = lspace:5 rspace:5 # short left tack
 operator.\u2ADF.infix = lspace:5 rspace:5 # short down tack
 operator.\u2AE0.infix = lspace:5 rspace:5 # short up tack
 operator.\u2AE1.infix = lspace:5 rspace:5 # perpendicular with s
 operator.\u2AE2.infix = lspace:5 rspace:5 # vertical bar triple right turnstile
 operator.\u2AE3.infix = lspace:5 rspace:5 # double vertical bar left turnstile
 operator.\u2AE4.infix = lspace:5 rspace:5 # &DoubleLeftTee;
 operator.\u2AE5.infix = lspace:5 rspace:5 # double vertical bar double left turnstile
@@ -1180,16 +1185,18 @@ operator.\u2AF7.infix = lspace:5 rspace:
 operator.\u2AF8.infix = lspace:5 rspace:5 # triple nested greater-than
 operator.\u2AF9.infix = lspace:5 rspace:5 # double-line slanted less-than or equal to
 operator.\u2AFA.infix = lspace:5 rspace:5 # double-line slanted greater-than or equal to
 operator.\u2AFB.infix = lspace:4 rspace:4 # triple solidus binary relation
 operator.\u2AFC.prefix = lspace:1 rspace:2 largeop movablelimits symmetric # large triple vertical bar operator
 operator.\u2AFD.infix = lspace:4 rspace:4 # double solidus operator
 operator.\u2AFE.infix = lspace:3 rspace:3 # white vertical bar
 operator.\u2AFF.prefix = lspace:1 rspace:2 largeop movablelimits symmetric # n-ary white vertical bar
+operator.\u2B45.infix = lspace:5 rspace:5 stretchy direction:horizontal # leftwards quadruple arrow
+operator.\u2B46.infix = lspace:5 rspace:5 stretchy direction:horizontal # rightwards quadruple arrow
 
 # Entries below are not part of the official MathML dictionary
 
 operator.\u0021.prefix = lspace:0 rspace:5 # !
 operator.\u0026.infix = lspace:5 rspace:5 # &amp;
 operator.\u0026.prefix = lspace:0 rspace:5 # &amp;
 operator.\u002B\u002B.prefix = lspace:0 rspace:2 # ++
 operator.\u002D\u002D.prefix = lspace:0 rspace:2 # --
